### PR TITLE
Add default zoom level (default 100%).

### PR DESCRIPTION
**Pull request by rysson**
Add config option `default_zoom` (default: 100) to adjust zoom factor on early part of game.

I don't know how to build DLL in my Linux, than I just change `Config.cs`, please check it. I've tested it by changing default value in `fetchConfig`.

BTW. When I added 
```javascript
const defaultZoom = constants.DEFAULT_ZOOM;
```
on top of `map.js` then setting was ignored in map `init()`. I've to use `constants.DEFAULT_ZOOM` directly.

Are you sure `exploreRadius` or `pixelSize` are taken from fetched config?
I've change `"explore_radius": 200` in my server config and add console message in `map.explore()`, it prints 100.

### CODE REVIEW

File: WebMap/Config.cs [hunk 1]
```diff
@@ -100,6 +101,12 @@ static class WebMapConfig {
             } catch {
                 System.Console.WriteLine("~~~ WebMap: FAILED TO PARSE cache_server_files VALUE IN CONFIG FILE AT: " + configFile + " . INVALID TYPE?");
             }
+
+            try {
+                DEFAULT_ZOOM = (int)configJson.GetValueOrDefault("default_zoom", 12);
```
Comment (kylepaulsen) [inline WebMap/Config.cs:106]: I think the 12 is supposed to be 100 here.

File: WebMap/Config.cs [hunk 2]
```diff
@@ -129,6 +136,7 @@ static class WebMapConfig {
             sb.Append($"\"pixel_size\":{PIXEL_SIZE},");
             sb.Append($"\"update_interval\":{PLAYER_UPDATE_INTERVAL},");
             sb.Append($"\"explore_radius\":{EXPLORE_RADIUS}");
+            sb.Append($"\"default_zoom\":{DEFAULT_ZOOM},");
```
Comment (kylepaulsen) [inline WebMap/Config.cs:139]: The line above this one needs a comma at the end. That's probably why the config wasn't being loaded correctly in the browser.

**Comment by kylepaulsen** [comment]
These changes look promising, I'll try and work them in next release.

File: WebMap/Config.cs (hunk 1 above)
Comment (rysson) [reply WebMap/Config.cs:106]: Sure, fixed.

File: WebMap/Config.cs (hunk 2 above)
Comment (rysson) [reply WebMap/Config.cs:139]: I missed, that JSON is build from hand. Fixed.

But as I wrote, I don't compile cs file, than I still use your DLL. I added log (for while) to see if exploreRadius is changed when I changed it in my server `config` file. Maybe I have config file in wrong place or the settings are not propagate.

```diff
--------------------------- WebMap/web-src/map.js ----------------------------
index 61b0fb6,61b0fb6..c12a768
@@@ -120,6 -120,6 +120,7 @@@ const redrawMap = () => 
  };
  
  const explore = (mapX, mapZ) => {
++	console.log('exploreRadius', exploreRadius);
  	const radius = exploreRadius / pixelSize;
  	const x = mapX / pixelSize + coordOffset;
  	const y = height - (mapZ / pixelSize + coordOffset);
```

I don't know how module global const is handled. The same change 
```javascript
const fetchConfig = fetch('/config').then(res => res.json()).then(config => {
  ...
  constants.DEFAULT_ZOOM = config.default_zoom || 200;   // 200 for testing only
```

Works with
```javascript
const init = (options) => {
  ...
  if (constants.DEFAULT_ZOOM != 100) ...
```

But **not** works with
```javascript
const defaultZoom = constants.DEFAULT_ZOOM;
...
const init = (options) => {
  ...
  if (defaultZoom != 100) ...
```

BTW config file should be in `plugins` or `config` in `Valheim/BepInEx`?

Comment (rysson) [reply WebMap/Config.cs:139]: The second case
```javascript
const defaultZoom = constants.DEFAULT_ZOOM;
const init = (options) => {
  console.log('zoom', defaultZoom, constants.DEFAULT_ZOOM);
```
shows
```
zoom 100 200
```

I know, that value is not created in JSON (old DLL) but I not sure that changes in `fetchConfig` (ex. zoom 200 for testing) is not propagated to `map` module. It seems the module-global const variables don't change after `fetchConfig`.

**Comment by rysson** [comment]
I make fixes, thank for review. I rebase it to current master,

**Comment by kylepaulsen** [comment]
Hey I wanted to fix my zoom code a bit, so I just merged in your changes with that. Also made the default zoom 200 as most of the time, people are probably going to be somewhere in the center. Anyway I just released which means this and #5 should be out now.

**Pull request closed without merging**